cf/service_access.go

```diff
@@ -8,48 +8,60 @@ import (
 	"net/http"
 	"net/url"
 
+	"github.com/Peripli/service-broker-proxy/pkg/platform"
+
 	"github.com/Peripli/service-manager/pkg/log"
 	cfclient "github.com/cloudfoundry-community/go-cfclient"
 	"github.com/pkg/errors"
 )
 
-// Metadata represents CF specific metadata that the proxy is concerned with.
-// It is currently used to provide context details for enabling and disabling of service access.
-type Metadata struct {
-	OrgGUID string `json:"organization_guid"`
-}
-
 // ServicePlanRequest represents a service plan request
 type ServicePlanRequest struct {
 	Public bool `json:"public"`
 }
 
 // EnableAccessForPlan implements service-broker-proxy/pkg/cf/ServiceVisibilityHandler.EnableAccessForPlan
 // and provides logic for enabling the service access for a specified plan by the plan's catalog GUID.
-func (pc *PlatformClient) EnableAccessForPlan(ctx context.Context, context json.RawMessage, catalogPlanID, platformBrokerName string) error {
-	return pc.updateAccessForPlan(ctx, context, catalogPlanID, platformBrokerName, true)
+func (pc *PlatformClient) EnableAccessForPlan(ctx context.Context, request *platform.ModifyPlanAccessRequest) error {
+	return pc.updateAccessForPlan(ctx, request, true)
 }
 
 // DisableAccessForPlan implements service-broker-proxy/pkg/cf/ServiceVisibilityHandler.DisableAccessForPlan
 // and provides logic for disabling the service access for a specified plan by the plan's catalog GUID.
-func (pc *PlatformClient) DisableAccessForPlan(ctx context.Context, context json.RawMessage, catalogPlanID, platformBrokerName string) error {
-	return pc.updateAccessForPlan(ctx, context, catalogPlanID, platformBrokerName, false)
+func (pc *PlatformClient) DisableAccessForPlan(ctx context.Context, request *platform.ModifyPlanAccessRequest) error {
+	return pc.updateAccessForPlan(ctx, request, false)
 }
 
-func (pc *PlatformClient) updateAccessForPlan(ctx context.Context, context json.RawMessage, catalogPlanID, platformBrokerName string, isEnabled bool) error {
-	metadata := &Metadata{}
-	if err := json.Unmarshal(context, metadata); err != nil {
-		return err
+type compositeError []error
+
+func (ce compositeError) Error() string {
+	result := ""
+	for _, e := range ce {
+		result += fmt.Sprintln(e.Error())
```

Comment:
There might be a lot of orgs and this will be a bit inefficient. Probably use `strings.Builder`

Reply:
I thought about this but writing to string builder returns an error which I can't return here

Reply:
Then maybe `strings.Join`? It would still be much better then string concatenation.